### PR TITLE
Add doc about tests with M1's

### PR DESCRIPTION
**Pull request by amegias**
It looks that Rosetta is the way to go with tests & M1's.
I've added a doc to know how to execute them / record them.

Please, I would like two check before integrating this doc:
1. Review the doc
2. Check if it works in your M1:
   1. Execute: `make setup` => It will install the dependencies (xcparse & xcbeautify)
   2. In main, execute: `arch -x86_64 make test` => **All tests passed**
   3. In `a118d23ff9754af48a96336043ee48c4cf43b051`, execute: `arch -x86_64 make test` => **361 tests failed** (like https://github.com/Telefonica/mistica-ios/actions/runs/4352377647/jobs/7605086596)

Thanks

### CODE REVIEW

**Review by DavidMarinCalleja — APPROVED** [review]
Test on an **Apple M1 Pro**, it works !!

 * terminal: 
Screenshot 2023-03-09 at 09 12 49
 * XCode: 
Screenshot 2023-03-09 at 09 23 12


Initially, I had two problems: 
- the first was that the _Brew_ dependencies were not installed, this was solved with `make setup`.
- the second error was that I still had the old `Mistica.xcworkspace` file and when I ran the `xcodebuild` command it didn't find any scheme. Solution: delete the _xcworkspace_ file.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -119,3 +119,21 @@ Open `Mistica.xcodeproj` after following the [How to work with mistica-ios](#how
 * Select the appropriate test target
 
 We use Screenshot or Snapshot type tests using a library named [SnapshotTesting](https://github.com/pointfreeco/swift-snapshot-testing), so please go the library repository and read the documentation.
+
+### Snapshot testing with Apple Silicon processors
+There are some slight differences in snapshots depending on whether they were generated by an Intel processor or Apple Silicon processor that cause the tests to fail (See more info [here](https://github.com/pointfreeco/swift-snapshot-testing/pull/628)).
+In order to avoid using `perceptualPrecision` where we could have false tests passing, we should do:
+#### Using the terminal
+* `make setup`: Needed to install the required dependencies.
```
Comment (amegias) [inline CONTRIBUTING.md:127]: @DavidMarinCalleja ;)

**Review by pbartolome — APPROVED** [review]
> make tests
`Tests Failed: 130 failed, 0 skipped, 410 total (52.483 seconds)`

> arch -x86_64 make test
`Tests Passed: 0 failed, 0 skipped, 410 total (88.092 seconds)`

**Review by alexanegon — COMMENTED** [review]
In my machine I have the following results:

m1-tests-doc:


a118d23ff9754af48a96336043ee48c4cf43b051:


main:

**Review by alejandroruizponce — COMMENTED** [review]
I did the test from main and failed 12 tests:

Screenshot 2023-03-09 at 17 00 37

**Comment by amegias** [comment]
@alejandroruizponce @alexanegon be sure your iPhone 14 - iOS 16.2 (what the _Makefile_ uses in `make test`) is in `en_US`. If you check the screenshots, the difference will be "Contactos" instead of "Contacts" (it already happened in my laptop :P )

**Comment by cbarcom** [comment]
I have same issue as @alexanegon and @alejandroruizponce and it was fixed changing simulator language as you suggested. Now all tests pass

**Review by cbarcom — APPROVED** [review]

**Comment by idenjoe** [comment]
Passed on main!

Screenshot 2023-03-10 at 13 32 45

**Review by idenjoe — APPROVED** [review]

**Comment by alexanegon** [comment]
I have finally managed to change the language of the simulator and everything is correct!
Captura de pantalla 2023-03-13 a las 9 22 07

**Pull request merged**